### DISCUSSION TITLE
Version 3.9 release?

### DISCUSSION THREAD

**Discussion by dsmmcken** [tradingview/lightweight-charts#1251] [General] [answered]
Are there plans for another version to be released? I see that there has been hundreds of commits to main since that last version 3.8 was released in Feb 2022.

**Comment by romfrancois** [comment, marked as answer]
Hi @dsmmcken !

You're absolutely right it's way overdue! A new version is right on the corner and should arrive later this week or the next.

**Comment by ThinksFast** [comment]
@romfrancois When do you expect a 4.0 release?  I'm looking to migrate to `lightweight-charts`, but need an easy way to theme the charts for light/dark color schemes, which I heard is a 4.0 feature 🙂🤞

**Reply by romfrancois** [reply]
We'll start the process internally this week as we have a few dependencies & checks to validate before officially releasing.

**Comment by SlicedSilver** [comment]
Version 4.0 has now been released 🎉
https://github.com/tradingview/lightweight-charts/releases/tag/v4.0.0

**Comment by ThinksFast** [comment]
Congrats @SlicedSilver and team!  Excited to try out 4.0.  

I've been looking through the documentation, but i couldn't find a page related to changing the chart's color scheme based on a light/dark theme.  Are there any examples out there that I could follow? 

RE: https://github.com/tradingview/lightweight-charts/issues/1001

**Reply by SlicedSilver** [reply]
The chart's colors are set via the options.

The customisation tutorial provide some good guidance on how to change the colors. https://tradingview.github.io/lightweight-charts/tutorials/customization/chart-colors

If you want to dynamically change the colors then you can use the `applyOptions` method to change the chart options at any point (and thus change the colors).

You may want to read this page on media queries: https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Testing_media_queries

You could place the `applyOptions` logic inside of an listener for the [prefers-color-scheme](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) media query if you want the chart color to match the devices current dark or light theme.

**Comment by ThinksFast** [comment]
@SlicedSilver thanks!  Is there a way to query all the charts on a page?  I'm not sure how to use `applyOptions` in a generic way.

For example, `chart.js` has helper functions to select all charts on a page and loop through them.  So this could be used to `applyOptions` programatically.  I don't see anything like this in the `lightweight-charts` docs.

```
Chart.helpers.each(Chart.instances, function(instance){
    // do something with each `instance`
} );

```

**Reply by SlicedSilver** [reply]
Lightweight charts doesn't have the concept of multiple charts. If you are creating multiple charts then you could store the IChartApi instance in an array or set, and loop over that array and apply the options.

If you are referring to Series then the same approach would work (when you create a Series, keep a reference to the instance in array).
